utilities/test_suite/rpp_test_suite_common.h

```diff
@@ -394,20 +577,105 @@ inline void convert_pkd3_to_pln3(Rpp8u *input, RpptDescPtr descPtr)
+        if (fileName.size() > 4 && fileName.substr(fileName.size() - 4) == ".jpg")
+        {
+            imageNamesPath.push_back(filePath);
+            imageNames.push_back(entity->d_name);
+        }
+    }
+    if(imageNames.empty())
+        std::cerr << "\n Did not load any file from " << folderPath;
+
+    closedir(src_dir);
+}
+
+// Searches for .jpg files in input folders
+void search_jpg_files(const string& folder_path, vector<string>& imageNames, vector<string>& imageNamesPath)
+{
+    vector<string> entry_list;
+    string full_path = folder_path;
+    auto sub_dir = opendir(folder_path.c_str());
+    if (!sub_dir)
+    {
+        std::cerr << "ERROR: Failed opening the directory at "<< folder_path << std::endl;
+        exit(0);
+    }
+
+    struct dirent* entity;
+    while ((entity = readdir(sub_dir)) != nullptr)
+    {
+        string entry_name(entity->d_name);
+        if (entry_name == "." || entry_name == "..")
+            continue;
+        entry_list.push_back(entry_name);
+    }
+    closedir(sub_dir);
+    std::sort(entry_list.begin(), entry_list.end());
+
+    for (unsigned dir_count = 0; dir_count < entry_list.size(); ++dir_count)
+    {
+        string subfolder_path = full_path + "/" + entry_list[dir_count];
+        fs::path pathObj(subfolder_path);
+        if (fs::exists(pathObj) && fs::is_regular_file(pathObj))
+        {
+            // ignore files with extensions .tar, .zip, .7z
+            auto file_extension_idx = subfolder_path.find_last_of(".");
+            if (file_extension_idx != std::string::npos)
+            {
+                std::string file_extension = subfolder_path.substr(file_extension_idx+1);
+                if ((file_extension == "tar") || (file_extension == "zip") || (file_extension == "7z") || (file_extension == "rar"))
+                    continue;
+            }
+            if (entry_list[dir_count].size() > 4 && entry_list[dir_count].substr(entry_list[dir_count].size() - 4) == ".jpg")
+            {
+                imageNames.push_back(entry_list[dir_count]);
+                imageNamesPath.push_back(subfolder_path);
+            }
+        }
+        else if (fs::exists(pathObj) && fs::is_directory(pathObj))
+            open_folder(subfolder_path, imageNames, imageNamesPath);
+    }
+}
+
+// Read a batch of images using the OpenCV library
+inline void read_image_batch_opencv(Rpp8u *input, RpptDescPtr descPtr, vector<string>::const_iterator imagesNamesStart)
 {
     for(int i = 0; i < descPtr->n; i++)
     {
         Rpp8u *inputTemp = input + (i * descPtr->strides.nStride);
-        string inputImagePath = imageNames[i];
+        string inputImagePath = *(imagesNamesStart + i);
         cv::Mat image, imageBgr;
         if (descPtr->c == 3)
         {
-            imageBgr = imread(inputImagePath, 1);
-            cvtColor(imageBgr, image, COLOR_BGR2RGB);
+            imageBgr = cv::imread(inputImagePath, 1);
+            cv::cvtColor(imageBgr, image, cv::COLOR_BGR2RGB);
```

Comment:
Why did we change instances to cv::
Can we not use namespace in the start?

Reply:
Done